doc/developers/contributing.rst

```diff
@@ -251,11 +251,18 @@ If any of the above seems like magic to you, then look up the `Git documentation
 web, or ask a friend or another contributor for help.
 
 If some conflicts arise between your branch and the ``master`` branch, you need
-to merge ``master``. The command will be::
+to merge ``master``. For that, you first need to fetch the ``upstream``, and
+then merge its ``master`` into your branch::
 
-  $ git merge master
+  $ git fetch upstream
+  $ git merge upstream/master
 
-with ``master`` being synchronized with the ``upstream``.
+It is recommended to work on temporary branches rather than directly on
```

Comment:
It can be a bit confusing for a first time contributor to read this if he already did point 5. hereabove. We should maybe say : "As explained in 5, it is recommended to work [....]. You can also create your feature branch from the latest ``upstream/master``:"

Reply:
And I would put this paragraph after the paragraph about solving conflicts so that the one about solving conflicts is just after the one explaining how to merge master.